File: 卡码/代码随想录/Go/图论/bfs/LeetCode_0463.go
// 463. 岛屿的周长
func islandPerimeter(grid [][]int) int {
	dir := [2][4]int{[4]int{0, 1, 0, -1}, [4]int{1, 0, -1, 0}}
	n, m := len(grid), len(grid[0])
	res := 0
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if grid[i][j] == 1 {
				for k := 0; k < 4; k++ {
					x, y := i+dir[0][k], j+dir[1][k]
					if x < 0 || x > n-1 || y < 0 || y > m-1 || grid[x][y] == 0 {
						res++
					}
				}
			}
		}
	}
	return res
}

File: 卡码/代码随想录/Python/图论/bfs/LeetCode_0463.py
#463. 岛屿的周长
from typing import *

class Solution:
  def islandPerimeter(self, grid: List[List[int]]) -> int:
    dir:List[List[int]]=[[0,1],[1,0],[0,-1],[-1,0]]
    n,m=len(grid),len(grid[0])
    res:int=0
    for i in range(n):
      for j in range(m):
        if grid[i][j]:
          for k in range(4):
            x,y=i+dir[k][0],j+dir[k][1]
            if x<0 or x>n-1 or y<0 or y>m-1 or grid[x][y]==0:
              res+=1
    return res